sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/ScalaReflectionSuite.scala

```diff
@@ -108,6 +108,16 @@ object TestingUDT {
   }
 }
 
+object TestingValueClass {
+  case class IntWrapper(i: Int) extends AnyVal
```

Comment:
does value class must be a case class?

Reply:
It doesn't but since Spark only supports `case class` (not `class`) for schema type. So I keep it that way.
Child columns can be `class` though. I think adding that in the future on top of this is not difficult.